src/store/modules/announcements.js

```diff
@@ -0,0 +1,51 @@
+import api from '../../api/api';
+
+export default {
+  namespaced: true,
+  state: {
+    announcements: [],
+  },
+  getters: {
+    getSitewideAnnouncements(state) {
+      return state.announcements.filter(
+        ann => !('event_id' in ann),
+      );
+    },
+    getEventAnnouncements(state, eventId) {
+      return state.announcements.filter(
+        ann => ('event_id' in ann) && ann.event_id === eventId,
+      );
+    },
+  },
+  mutations: {
+    loadAnnouncements(state, { announcements }) {
+      announcements.forEach((ann) => {
+        if (state.announcements.filter(a => a.id === ann.id).length === 0) {
```

Comment:
This seems a little overcomplicated/inefficient. There should be a contains function we can use?

Reply:
I had trouble getting that working. I tried doing this: 
![image](https://user-images.githubusercontent.com/20273689/90991442-1be3bc00-e577-11ea-9ac3-c18804a3c4be.png)

but the conditional always returned true and so whenever something new gets loaded, all the old announcements stay and get duplicated. I'm not familiar with the technicalities of how javascript checks for equality. Do we have to override something like equals() and hashCode() in Java?

Reply:
I think we might want to store data in Vuex differently so that we don't need this logic. Storing all announcements in one array requires us to have this logic to create that array, and logic in the getters to filter the array. A better solution might be storing sitewide announcements and event announcements separately.

Reply:
Gotcha. I can change that